File: backend/src/main/java/com/back/fairytale/domain/like/entity/Like.java
package com.back.fairytale.domain.like.entity;

import com.back.fairytale.domain.fairytale.entity.Fairytale;
import com.back.fairytale.domain.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Builder
@Table(name = "likes", indexes = {
        @Index(name = "idx_like_user_fairytale", columnList = "user_id, fairytale_id")
})
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Like {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "fairytale_id", nullable = false)
    private Fairytale fairytale;

    public static Like toEntity(User user, Fairytale fairytale) {
        return Like.builder()
                .user(user)
                .fairytale(fairytale)
                .build();
    }

}

File: backend/src/main/java/com/back/fairytale/domain/like/entity/Like.kt
package com.back.fairytale.domain.like.entity

import com.back.fairytale.domain.fairytale.entity.Fairytale
import com.back.fairytale.domain.user.entity.User
import jakarta.persistence.*

@Entity
@Table(
    name = "likes",
    indexes = [
        Index(name = "idx_like_user_fairytale", columnList = "user_id, fairytale_id")
    ]
)
class Like(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val user: User,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "fairytale_id", nullable = false)
    val fairytale: Fairytale
) {
    companion object {
        fun toEntity(user: User, fairytale: Fairytale): Like {
            return Like(
                user = user,
                fairytale = fairytale
            )
        }
    }
}